### PR TITLE
Update RBC Royal Bank - Add RBC Caribbean

### PR DESCRIPTION
**Pull request by 40504**
I'm not sure what type of 2fa is used, please check the documentation page :
https://www.rbcroyalbank.com/caribbean/digital-hub/security-two-factor.html

### CODE REVIEW

**Review by ApCoder123 — CHANGES_REQUESTED** [review]
From the links provided, it looks like 2FA for RBC is only available in the Caribbean. 

Given that this is the case, I think it makes sense to break the RBC entries into one without 2FA and RBC Caribbean with 2FA, similar to this:
image

- Please create a new entry under `entries/c/caribbean.rbcroyalbank.com`
- Make sure to add the region codes for the countries listed below.
- image
```JSON
"regions": [
  "ag",
  "aw",
  "bs",
  "tc",
  "bq",
  "ky",
  "cw",
  "dm",
  "gd",
  "ms",
  "kn",
  "lc",
  "sx",
  "vc",
  "tt"
]
```

- The 2FA type would be `custom-software` based on the description of what is needed to create a Soft Token:

> The Advice Centre representative will provide you with the information needed to setup your Soft Token including:
> Name: RBC User ID
> Serial Number
> Activation Code

```JSON
"custom-software": [
  "Entrust Identity Guard"
]
```

- It looks like `custom-hardware` is also available, but only for business customers (There should be a `notes` field to mention this)

Thanks.

File: entries/c/caribbean.rbcroyalbank.com.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "RBC Caribbean": {
+    "domain": "caribbean.rbcroyalbank.com",
+    "img": "rbcroyalbank.com.svg",
+    "tfa": [
+      "custom-software"
+    ],
+    "keywords": [
```
Comment (ApCoder123) [inline entries/c/caribbean.rbcroyalbank.com.json]:
```suggestion
    "regions": [
```

File: entries/c/caribbean.rbcroyalbank.com.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      "cw",
+      "dm",
+      "gd",
+      "ms",
+      "kn",
+      "lc",
+      "sx",
+      "vc",
+      "tt"
+    ]
```
Comment (ApCoder123) [inline entries/c/caribbean.rbcroyalbank.com.json:33]:
```suggestion
    ],
    "keywords": [
      "banking"
    ]
```

File: entries/c/caribbean.rbcroyalbank.com.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "RBC Caribbean": {
+    "domain": "caribbean.rbcroyalbank.com",
+    "img": "rbcroyalbank.com.svg",
+    "tfa": [
+      "custom-software"
```
Comment (ApCoder123) [inline entries/c/caribbean.rbcroyalbank.com.json]:
```suggestion
      "custom-software",
      "custom-hardware"
```

File: entries/c/caribbean.rbcroyalbank.com.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    ],
```
Comment (ApCoder123) [inline entries/c/caribbean.rbcroyalbank.com.json:8]:
```suggestion
    ],
    "custom-software": [
      "Entrust Identity Guard"
    ],
    "notes": "Hardware 2FA methods are only available for business customers",
    "documentation": "https://www.rbcroyalbank.com/caribbean/digital-hub/security-two-factor.html",
```

**Review by ApCoder123 — CHANGES_REQUESTED** [review]

**Review by ApCoder123 — APPROVED** [review]

**Review by phallobst — APPROVED** [review]

**Pull request merged**